common/kvstore/src/test/java/org/apache/spark/util/kvstore/ArrayKeyIndexType.java

```diff
@@ -38,7 +38,6 @@ public boolean equals(Object o) {
 
   @Override
   public int hashCode() {
-    return key.hashCode();
+    return Arrays.hashCode(key);
```

Comment:
This is fixing existing bug - without fixing this, comparison of both Sets/Maps which contain ArrayKeyIndexType as key would fail.

Reply:
Just created a separate PR for this: #26709